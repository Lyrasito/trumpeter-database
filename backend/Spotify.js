const fetch = require("node-fetch");
const dotenv = require("dotenv");
dotenv.config();

const getSpotifyToken = async () => {
  const response = await fetch("https://accounts.spotify.com/api/token", {
    method: "post",
    headers: {
      "Content-type": "application/x-www-form-urlencoded",
      Authorization:
        "Basic " +
        new Buffer(
          process.env.SPOTIFY_ID + ":" + process.env.SPOTIFY_SECRET
        ).toString("base64"),
    },
    body: "grant_type=client_credentials",
  });
  const jsonResponse = await response.json();
  return jsonResponse.access_token;
};

const getAlbum = (term) => {
  const accessToken = await getSpotifyToken();
  try {
    const response = await fetch(
      `https://api.spotify.com/v1/search?q=${term}&type=album`,
      {
        headers: {
          Authorization: "Bearer " + accessToken,
        },
      }
    );
    const jsonResponse = await response.json();
    const album = jsonResponse.albums.items[0];
    return album;
  } catch (err) {
    console.log(err);
  }
}
const getPlayer = async (name) => {
  const accessToken = await getSpotifyToken();
  try {
    const response = await fetch(
      `https://api.spotify.com/v1/search?q=${name}&type=artist`,
      {
        headers: {
          Authorization: "Bearer " + accessToken,
        },
      }
    );
    const jsonResponse = await response.json();
    const player = jsonResponse.artists.items[0];
    return player;
  } catch (err) {
    console.log(err);
  }
}

module.exports = {getSpotifyToken, getAlbum, getPlayer};
