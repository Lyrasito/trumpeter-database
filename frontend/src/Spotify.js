import fetch from "node-fetch";

if (isHeroku) {
  baseUrl = "https://trumpeter-database-backend.herokuapp.com/api";
} else {
  baseUrl = "http://localhost:4000/api";
}

const Spotify = {
  async getAccessToken() {
    const response = await fetch(`${baseUrl}/token`);
    const jsonResponse = await response.json();
    return jsonResponse.token;
  },

  async searchAlbum(term) {
    const accessToken = await Spotify.getAccessToken();
    try {
      const response = await fetch(
        `https://api.spotify.com/v1/search?q=${term}&type=album`,
        {
          headers: {
            Authorization: "Bearer " + accessToken,
          },
        }
      );
      const jsonResponse = await response.json();
      const id = jsonResponse.albums.items[0].id;
      return `https://open.spotify.com/album/${id}`;
    } catch (err) {
      console.log(err);
    }
  },
  async getAlbum(term) {
    const accessToken = await Spotify.getAccessToken();
    try {
      const response = await fetch(
        `https://api.spotify.com/v1/search?q=${term}&type=album`,
        {
          headers: {
            Authorization: "Bearer " + accessToken,
          },
        }
      );
      const jsonResponse = await response.json();
      const album = jsonResponse.albums.items[0];
      return album;
    } catch (err) {
      console.log(err);
    }
  },
  async getPlayer(name) {
    const accessToken = await Spotify.getAccessToken();
    try {
      const response = await fetch(
        `https://api.spotify.com/v1/search?q=${name}&type=artist`,
        {
          headers: {
            Authorization: "Bearer " + accessToken,
          },
        }
      );
      const jsonResponse = await response.json();
      const player = jsonResponse.artists.items[0];
      return player;
    } catch (err) {
      console.log(err);
    }
  },
};

export default Spotify;
