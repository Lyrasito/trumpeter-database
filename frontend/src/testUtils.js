const fakePlayer = () => ({
  id: 1,
  name: "Fake Player",
  city: "Fake City",
  startYear: 1234,
  endYear: 5678,
});

const fakePlayers = () => [
  {
    id: 1,
    name: "Fake Player",
    city: "Fake City",
    startYear: 1234,
    endYear: 5678,
  },
  {
    id: 2,
    name: "Fake Player 2",
    city: "Fake City 2",
    startYear: 2345,
    endYear: 6789,
  },
];

const fakeAlbums = () => [
  {
    id: 1,
    title: "Fake Album 1",
    year: 1234,
    genre: "fakeGenre1",
    player_id: 1,
  },
  {
    id: 2,
    title: "Fake Album 2",
    year: 1234,
    genre: "fakeGenre2",
    player_id: 1,
  },
];

const fakeAlbum = () => ({
  id: 1,
  title: "Fake Album 1",
  year: 1234,
  genre: "fakeGenre1",
  player_id: 1,
});

const fakeGenres = () => ["fakeGenre1", "fakeGenre2"];

export { fakeAlbum, fakeAlbums, fakePlayer, fakeGenres, fakePlayers };
