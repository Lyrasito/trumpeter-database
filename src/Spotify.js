const fetch = require("node-fetch");
let accessToken;
let userID;
const request = require("request");
const clientID = "client-id";
const clientSecret = "client-secret";
const redirectURI = "http://localhost:3000/";

const Spotify = {
  getAccessToken() {
    var authOptions = {
      url: "https://accounts.spotify.com/api/token",
      headers: {
        Authorization:
          "Basic " +
          new Buffer(clientID + ":" + clientSecret).toString("base64"),
      },
      form: {
        grant_type: "client_credentials",
      },
      json: true,
    };

    request.post(authOptions, function (error, response, body) {
      if (!error && response.statusCode === 200) {
        accessToken = body.access_token;
        return accessToken;
      }
    });
  },

  async searchAlbum(term) {
    const response = await fetch(
      `https://api.spotify.com/v1/search?q=${term}&type=album`,
      {
        headers: {
          Authorization: "Bearer" + accessToken,
        },
      }
    );
    const jsonResponse = await response.json();
    console.log(jsonResponse);
  },
};
Spotify.getAccessToken();
Spotify.searchAlbum("Birth of the cool");
