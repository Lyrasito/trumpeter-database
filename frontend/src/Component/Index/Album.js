import React from "react";
import Spotify from "../../Spotify";

class Album extends React.Component {
  state = {
    link: "",
  };

  componentDidMount() {
    this.getAlbumLink();
  }

  getAlbumLink = async () => {
    let link = await Spotify.searchAlbum(
      `${this.props.album.title} ${this.props.player.name}`
    );
    this.setState({
      link,
    });
  };

  render() {
    const { album } = this.props;
    return (
      <div key={album.id} className="album">
        <img
          src={album.image ? album.image : "./img/album-default.png"}
          className="album-image"
          alt={album.title}
        />
        <h5 className="album-title">{album.title} </h5>{" "}
        <label>{album.year}</label>
        <br />
        <label>Genre: {album.genre}</label>
        <a
          href={this.state.link}
          className="link"
          target="_blank"
          rel="noopener noreferrer"
        >
          <p>Open with Spotify</p>
        </a>
      </div>
    );
  }
}

export default Album;

//
