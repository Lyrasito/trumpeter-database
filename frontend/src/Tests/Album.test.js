import Album from "../Component/Index/Album";
import React from "react";
import toJSON from "enzyme-to-json";
import { shallow } from "enzyme";
import wait from "waait";
import Spotify from "../Spotify";
import { fakePlayer, fakeAlbum } from "../testUtils";

const player = fakePlayer();
const album = fakeAlbum();

Spotify.searchAlbum = jest.fn().mockResolvedValue("fakespotifylink.com");

describe("<Album />", () => {
  it("renders properly", async () => {
    const wrapper = shallow(<Album player={player} album={album} />);
    await wait();
    wrapper.update();

    expect(wrapper.find("[href='fakespotifylink.com']")).toHaveLength(1);
    expect(toJSON(wrapper)).toMatchSnapshot();
  });
});
