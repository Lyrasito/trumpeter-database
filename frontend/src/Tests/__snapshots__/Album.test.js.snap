// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Album /> renders properly 1`] = `
<div
  className="album"
  key="1"
>
  <img
    alt="Fake Album 1"
    className="album-image"
    src="./img/album-default.png"
  />
  <h5
    className="album-title"
  >
    Fake Album 1
     
  </h5>
   
  <label>
    1234
  </label>
  <br />
  <label>
    Genre: 
    fakeGenre1
  </label>
  <a
    className="link"
    href="fakespotifylink.com"
    rel="noopener noreferrer"
    target="_blank"
  >
    <p>
      Open with Spotify
    </p>
  </a>
</div>
`;
